micro-rdk/src/common/ota.rs

```diff
@@ -334,71 +327,108 @@ impl<S: OtaMetadataStorage> OtaService<S> {
+                    Ok(connection) => {
+                        match connection.await {
+                            Ok(io) => {
+                                match http2::Builder::new(self.exec.clone())
+                                    .max_frame_size(16_384) // lowest configurable
+                                    .timer(H2Timer)
+                                    .handshake(io)
+                                    .await
+                                {
+                                    Ok(pair) => break pair,
+                                    Err(e) => {
+                                        log::error!("failed to build http request: {}", e);
+                                    }
                                 }
                             }
-                        }
-                        Err(e) => {
-                            log::error!("failed to create tcp stream: {}", e);
+                            Err(e) => {
+                                log::error!("failed to create tcp stream: {}", e);
+                            }
                         }
                     }
-                }
-                Err(e) => {
-                    log::error!("failed to create http connection: {}", e);
-                }
-            };
+                    Err(e) => {
+                        log::error!("failed to create http connection: {}", e);
+                    }
+                };
 
-            log::debug!(
-                "retry establishing http connection to {} in {} seconds",
-                &uri,
-                CONN_RETRY_SECS
-            );
-            Timer::after(Duration::from_secs(CONN_RETRY_SECS)).await;
-        };
+                log::debug!(
+                    "retry establishing http connection to {} in {} seconds",
+                    &uri,
+                    CONN_RETRY_SECS
+                );
+                Timer::after(Duration::from_secs(CONN_RETRY_SECS)).await;
+            };
 
-        let conn = Box::new(self.exec.spawn(async move {
-            if let Err(err) = conn.await {
-                log::error!("connection failed: {:?}", err);
+            let conn = Box::new(self.exec.spawn(async move {
+                if let Err(err) = conn.await {
+                    log::error!("connection failed: {:?}", err);
+                }
+            }));
+
+            log::info!("ota connected, beginning download");
+            let request = Request::builder()
+                .method("GET")
+                .uri(uri)
+                .body(Empty::<Bytes>::new())
+                .map_err(|e| OtaError::Other(e.to_string()))?;
+            response = sender
+                .send_request(request)
+                .await
+                .map_err(|e| OtaError::Other(e.to_string()))?;
+
+            // TODO(RSDK-9346): handle other success codes
+            let status = response.status();
+            match (status.is_success(), status.is_redirection()) {
```

Comment:
can switch this for a `if-else` chain if preferable

Reply:
I think the `match` is fine.